sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -1190,6 +1199,12 @@ pi_result piEnqueueMemBufferRead(pi_queue Queue, pi_mem Src,
 
   _pi_buffer *buf = static_cast<_pi_buffer *>(Src);
 
+  std::unique_ptr<_pi_event> RetEv{nullptr};
+  if (Event) {
+    RetEv = std::unique_ptr<_pi_event>(new _pi_event());
+    RetEv->IsDummyEvent = true;
```

Comment:
As I understand your current implementation, with all commands being blocking, all events returned from enqueue operations are effectively in "completed" state once control is returned back to the caller of the enqueue operation.

- Given that I don't see the need in IsDummyEvent field.
- Second, I don't see the need in using `unique_ptr` here, as there is no chance RetEv will escape to any other thread
- Third, the above code is not needed, instead, the line 1218 should be '*Event = new _pi_event();' plus maybe setting the event to 'completed' state.

The same applies to other places where events are created.

Reply:
@romanovvlad suggested this implementation from CUDA PI implementation from recent phone call. Also, he was fine with this 'DummyEvent' approach.

Reply:
> @romanovvlad suggested this implementation from CUDA PI implementation from recent phone call. 

OK, now I see the point - unique_ptr is useful to automatically release the memory allocated by new _pi_even in case buf->CmBufferPtr->ReadSurface throws an exception. Disregard this my comment then.

> Also, he was fine with this 'DummyEvent' approach.
`DummyEvent` seems is redundant, as I explained above, even though harmless. So I don't mind leaving it as is for now.